src/auth/entities/user.entity.ts

```diff
@@ -0,0 +1,108 @@
+import { Optional } from 'sequelize';
+import {
+  Table,
+  Model,
+  Column,
+  DataType,
+  HasOne,
+  HasMany,
+} from 'sequelize-typescript';
+import { Roles } from '../enums/roles-enum';
+import { RegistrationMethod } from '../enums/registMethod-enum';
+import { VerificationCodes } from './verify.entity';
+import { Jwt } from './jwt.entity';
+import { UserAddress } from 'src/user/entities/address.entity';
+import { ApiProperty } from '@nestjs/swagger';
+
+export interface PersonAttributes {
+  id: number;
+  name: string;
+  lastname: string;
+  email: string;
+  numbers: string;
+  password: string;
+  role: string;
+  token: string;
+  isVerified: boolean;
+}
+
+export interface PersonCreationAttributes
+  extends Optional<
+    PersonAttributes,
+    'id' | 'role' | 'token' | 'isVerified' | 'name' | 'lastname'
+  > {}
+
+@Table({
+  timestamps: true,
+  tableName: 'users',
+  paranoid: true,
+  schema: 'store',
+})
+export class User extends Model<PersonAttributes, PersonCreationAttributes> {
+  @Column({ type: DataType.STRING(32), unique: false, allowNull: true })
+  name?: string;
+
+  @Column({ type: DataType.STRING(52), unique: false, allowNull: true })
+  lastname?: string;
+
+  @Column({
+    type: DataType.STRING(32),
+    unique: true,
+    allowNull: true,
```

Comment:
Why all the fields and especially email are `allowNull: true`? How can be user without email as the identifier?

Reply:
Because a user can register by email or phone number, and one of these fields can be blank. If by phone number, the email can be blank, and vice versa

Reply:
Oh, got it. Thanks.